framework/auth/views.py

```diff
@@ -637,3 +639,79 @@ def resend_confirmation_post(auth):
 
     # Don't go anywhere
     return {'form': form}
+
+
+def oauth_user_email_get():
+    """
+    Landing view for first-time oauth-login user to enter their email address.
+    HTTP Method: GET
+    """
+
+    form = ResendConfirmationForm(request.form)
+    session = get_session()
+    if not session.is_oauth_first_time_login:
+        raise HTTPError(http.UNAUTHORIZED)
+
+    oauth_provider = session.data['oauth_user_provider']
+
+    return {
+        'form': form,
+        'oauth_provider': oauth_provider.upper()
+    }
+
+
+def oauth_user_email_post():
+    """
+    View to handle email submission for first-time oauth-login user.
+    HTTP Method: POST
+    """
+
+    form = ResendConfirmationForm(request.form)
+    session = get_session()
+    if not session.is_oauth_first_time_login:
+        raise HTTPError(http.UNAUTHORIZED)
+
+    oauth_provider = session.data['oauth_user_provider']
+    oauth_id = session.data['oauth_user_id']
+    oauth_fullname = session.data['oauth_user_fullname']
+
+    if form.validate():
+        clean_email = form.email.data
+        user = get_user(email=clean_email)
+        oauth = {oauth_provider: oauth_id}
+        if user:
+            # TODO: link user's OSF account with ORCID
```

Comment:
The ordering for this `TODO` is off -- as discussed, this could lead to ORCID logins gaining access to any OSF account, so the confirmation email needs to be step 1, with following that link in that email triggering the rest.